TELCODOCS-236: 4.8 Release Notes

jcfrye77 opened the pull request:
1st draft

Preview: https://deploy-preview-33359--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-8-release-notes.html#ocp-4-8-bug-fixes

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
 [id="ocp-4-8-bug-fixes"]
 == Bug fixes
 
+*Bare Metal Hardware Provisioning*
```
Just keep adding to the equal signs:

```
[id="ocp-4-8-bug-fixes-bare-metal-hardware-provisioning"]
=== Bare metal hardware provisioning
```

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
 
+*Bare Metal Hardware Provisioning*
+
+* Previously, Ironic failed to download an image for installation because it uses HTTPS by default and Ironic does not have the correct certificate bundle available. This issue is fixed by setting the image download as insecure to request a transfer without the certificate. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1953795[*BZ#1953795*])
```
Restate the subject--it gets restated after "it" and that _could_ cause confusion.
s/it/Ironic/

Continue with the past tense that is already in use:
s/does not/did not/

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
+
+* Previously, Ironic failed to download an image for installation because it uses HTTPS by default and Ironic does not have the correct certificate bundle available. This issue is fixed by setting the image download as insecure to request a transfer without the certificate. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1953795[*BZ#1953795*])
+
+* Previously, when using dual stack deployments, worker node hostnames sometimes did not match the name inspected before deployment. This caused nodes to need manual approval. This has been fixed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1955114[*BZ#1955114*])
```
s/dual stack deployments/dual-stack networking/

s/hostnames/host names/

"the name inspected before deployment" seems a little off.  Can you indicate what did the inspecting?..."...sometimes did not match the host name that <xyz> inspected/determined before installation."

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
+
+* Previously, when using dual stack deployments, worker node hostnames sometimes did not match the name inspected before deployment. This caused nodes to need manual approval. This has been fixed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1955114[*BZ#1955114*])
+
+* Previously, in UEFI mode, the ironic-python-agent created a UEFI bootloader entry after downloading the RHCOS image. When using an RHCOS image based on RHEL 8.4, the image could fail to boot using this entry. If the entry installed by Ironic was used when booting the image, the boot could fail and output a BIOS error screen. This is fixed by the ironic-python-agent configuring the boot entry ba ...
```
maybe `ironic-python-agent` in backticks both places) because it sounds like the name of a daemon or service.  See the [Quick reference](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference) section of the docs guidelines.

Maybe "...after the agent downloads the RHCOS image."--to be crystal clear on what performs the action.

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
+
+* Previously, in UEFI mode, the ironic-python-agent created a UEFI bootloader entry after downloading the RHCOS image. When using an RHCOS image based on RHEL 8.4, the image could fail to boot using this entry. If the entry installed by Ironic was used when booting the image, the boot could fail and output a BIOS error screen. This is fixed by the ironic-python-agent configuring the boot entry ba ...
+
+* Previously, a node sometimes selected the incorrect IP version upon startup (v6 instead of v4, or vice versa). The node would fail to come up because it did not receive an IP address. This is fixed by cluster-baremetal-operator passing the IP Option to the downloader (ip=dhcp or ip=dhcp6), so this is set correctly at startup and the node comes up as expected. (link:https://bugzilla.redhat.com/s ...
```
Please use IPv4 and IPv6--it's not mysterious without, it's just consistent.

Please use "fail to start because" instead of "fail to come up because".  Formality isn't bad and it *might* translate better.

Operator names are treated as [proper nouns](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#operator-name-capitalization) according to the guidelines, so I expected "the Cluster Bare Metal Operator" instead of "cluster-baremetal-operator".

Please lowercase "option" so we get "IP option" and also enclose each of the sample arguments in backticks to format them as [inline code](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#inline-code-or-commands) like "...(`ip=dhcp` or `ip=dhcp6`),..."

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
+
+* Previously, a node sometimes selected the incorrect IP version upon startup (v6 instead of v4, or vice versa). The node would fail to come up because it did not receive an IP address. This is fixed by cluster-baremetal-operator passing the IP Option to the downloader (ip=dhcp or ip=dhcp6), so this is set correctly at startup and the node comes up as expected. (link:https://bugzilla.redhat.com/s ...
+
+*Assisted Installer*
```
Same feedback as earlier, please use equal signs to make this a heading and add an ID.  Support folks want to be able to send a link to a specific heading.

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
+
+*Assisted Installer*
+
+* Previously, assisted-service did not wait for postgres to start up and be ready to accept connections. Assisted-service attempted to establish a database connection, failed, and assisted-service container failed and restarted. This issue has been fixed by assisted-service attempting to connect to the database for up to 10 seconds. If postgres starts and is ready to accept connection within 10 s ...
```
My sugg is to treat the container names like daemon or system names and go with something like "Previously, the `assisted-service` container...".  The same for `postgres`, because otherwise, you should probably use the project name, PostgreSQL and that is probably not what you want to do.

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
+
+* Previously, assisted-service did not wait for postgres to start up and be ready to accept connections. Assisted-service attempted to establish a database connection, failed, and assisted-service container failed and restarted. This issue has been fixed by assisted-service attempting to connect to the database for up to 10 seconds. If postgres starts and is ready to accept connection within 10 s ...
+
+*RHCOS*
```
Please use heading markup like the previous headings.

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -193,6 +193,24 @@ In the table, features are marked with the following statuses:
+
+*RHCOS*
+
+* Previously, the SELinux policy was not allowing systemd to read files under /etc/kubernetes, and the kubelet failed to start. This is fixed by updating the SELinux policy to allow systemd to read files with kubernetes_file_t labels. The kubelet starts successfully. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1969998[*BZ#1969998*])
```
The systemd process has some kind of immunity from being formatted as a daemon--I have no idea why.  Please format `/etc/kubernetes` in backticks because it is a directory and formatting directories is covered in the [quick reference](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference) section of the guidelines.

Kubelet also seems to have the same immunity from formatting that systemd has.

If `kubernetes_file_t` is a system term or item as identified in the quick reference, then please enclose it in backticks.

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -409,6 +427,10 @@ This script removes unauthenticated subjects from the following cluster role bin
 +
 (link:https://bugzilla.redhat.com/show_bug.cgi?id=1821771[*BZ#1821771*])
 
+*Networking*
```
Please format this as a title and add an ID like other titles.

mikemckiernan commented on release_notes/ocp-4-8-release-notes.adoc:
```diff
@@ -409,6 +427,10 @@ This script removes unauthenticated subjects from the following cluster role bin
 
+*Networking*
+
+* When running CNF tests on SNO and regular mode, the logic in place to understand if the cluster is ready is missing details. In particular, creating an SR-IOV network will not create a network attachment definition until at least one minute elapses. All the DPDK tests fail in cascade. Run the CNF tests skipping the DPDK feature when running against SNO in regular mode, with the -ginkgo.skip par ...
```
Please expand SNO unless you feel confident that customers know what that is.

Maybe instead of "the logic in place to understand" can be "the tests failed to determine when the cluster is ready".  That might not be quite correct.  If you can make the sentence less wordy, please do.  

The use of "in cascade" might not be necessary and might not translate well.  I suggest removing it.

Please enclose `-ginko.skip` in backticks as suggested in the [quick reference](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference) section of the guidelines.

mikemckiernan reviewed (changes requested):
Please let me know if I can clarify any of my comments.

jcfrye77 commented:
@mikemckiernan Thanks a million for the quick and extremely helpful review! I am learning a tremendous amount from your comments, thank you.

jcfrye77 commented:
@mikemckiernan Thanks again for the brilliant review on Friday. I've incorporated all the changes, and also added two new RNs. Thank you for all your time and help.
